### PR TITLE
WIP: add basics for static method handling

### PR DESCRIPTION
**Pull request by Idrinth**
refs #29
likely going to break psr-2 compatibility!

### CODE REVIEW

File: src/Implementations/MethodDescriptor.php [hunk 1]
```diff
@@ -32,14 +37,16 @@ class MethodDescriptor implements MDI
      * @param string $name
      * @param Type[] $params
      * @param Type $return
+     * @param boolean $static
      * @param ClassType[] $exceptions
      */
-    public function __construct($name, $params, Type $return, $exceptions = array())
+    public function __construct($name, $params, Type $return, $static = false, $exceptions = array())
```
Comment (JEM94) [inline src/Implementations/MethodDescriptor.php:43]: public function __construct($name, array $params, Type $return, $static = false, array $exceptions = array())

File: templates/static-type-test-wrapper.twig [hunk 2]
```diff
@@ -0,0 +1,3 @@
+$return = {{class}}::{{function.name}}{% include 'function.twig' with {'function': function} only %};
+
+{% include 'type-tests.twig' with {'return': function.return, 'variable': '$return'} only %}
```
Comment (JEM94) [inline templates/static-type-test-wrapper.twig:3]: no new line at end of file, but it's a twig template so it doesn't matter, right?

File: src/Implementations/MethodDescriptor.php (hunk 1 above)
Comment (JEM94) [reply src/Implementations/MethodDescriptor.php:43]: but.. this should work in lower php versiosn too? then: don't care about this

File: src/Implementations/MethodDescriptor.php [hunk 3]
```diff
@@ -23,6 +23,11 @@ class MethodDescriptor implements MDI
      */
     private $return;
 
+    /**
+     * @var boolean
+     */
+    private $static;
+
     /**
      * @var Type[]
```
Comment (JEM94) [inline src/Implementations/MethodDescriptor.php:32]: ClassType

File: src/Interfaces/MethodDescriptor.php [hunk 4]
```diff
@@ -19,6 +19,11 @@ public function getParams();
      */
     public function getReturn();
 
+    /**
+     * @return boolean
+     */
+    public function isStatic();
+
     /**
      * @return Type[]
```
Comment (JEM94) [inline src/Interfaces/MethodDescriptor.php:28]: ClassType again pls

File: src/Implementations/MethodDescriptor.php (hunk 1 above)
Comment (Idrinth) [reply src/Implementations/MethodDescriptor.php:43]: `bool $static` is not a option, should work with php5 after ally

File: src/Interfaces/MethodDescriptor.php (hunk 4 above)
Comment (Idrinth) [reply src/Interfaces/MethodDescriptor.php:28]: why would a method only be allowed to return a class?

File: src/Implementations/MethodDescriptor.php (hunk 3 above)
Comment (Idrinth) [reply src/Implementations/MethodDescriptor.php:32]: there is no reason why there shouldn't be none-classparameters for example

File: templates/static-type-test-wrapper.twig (hunk 2 above)
Comment (Idrinth) [reply templates/static-type-test-wrapper.twig:3]: yes, would also mess up the output

**Review by JEM94 — APPROVED** [review]
nevermind :D lgtm

**Pull request merged**